client/sniffer/src/test/java/org/elasticsearch/client/sniff/SnifferTests.java

```diff
@@ -653,4 +654,23 @@ public void testDefaultSchedulerShutdown() throws Exception {
         verify(executor, times(2)).awaitTermination(1000, TimeUnit.MILLISECONDS);
         verifyNoMoreInteractions(executor);
     }
+
+    static final void assertNodesEquals(List<Node> expected, List<Node> actual) {
+        try {
+            assertEquals(expected.size(), actual.size());
+            Iterator<Node> expectedItr = expected.iterator();
+            Iterator<Node> actualItr = actual.iterator();
+            while (expectedItr.hasNext()) {
+                Node expectedNode = expectedItr.next();
+                Node actualNode = actualItr.next();
+                assertEquals(expectedNode.getHost(), actualNode.getHost());
+                assertEquals(expectedNode.getBoundHosts(), actualNode.getBoundHosts());
+                assertEquals(expectedNode.getName(), actualNode.getName());
+                assertEquals(expectedNode.getVersion(), actualNode.getVersion());
+                assertEquals(expectedNode.getRoles(), actualNode.getRoles());
```

Comment:
maybe we should have Node implement equals and hashCode?

Reply:
I had that but you didn't like it because it implements equals for all the parts of Node but you thought that really only the host was important. Which is true, but it means we have to do this here.

Reply:
yea I remember now, sorry :)

Reply:
if you prefer moving this back to Node I am fine with it, no biggie.

Reply:
Yeah, I like having it in node more. I'll move it back.